httpclient5/src/main/java/org/apache/hc/client5/http/config/ConnectionConfig.java

```diff
@@ -149,9 +149,19 @@ public Builder setSocketTimeout(final int soTimeout, final TimeUnit timeUnit) {
         }
 
         /**
-         * Determines the default socket timeout value for I/O operations.
+         * Determines the default socket timeout value for I/O operations on
+         * connections created by this configuration.
+         * A timeout value of zero is interpreted as an infinite timeout.
          * <p>
-         * Default: {@code null} (undefined)
+         * This value serves as the baseline timeout applied when a connection is first
+         * created and managed by the connection pool.
+         * </p>
+         * <p>
+         * Please note: in classic (blocking) I/O execution, if a request-level
```

Comment:
@LichKing-lee I do not think this is the right place to mention any specific i/o layers or anything that the protocol may or may not do. `RequestConfig` is of no relevance here. What you can mention here instead is that the socket  timeout when specified at the connection layer overrides the default set by the i/o layer and in its turn can be overridden by the protocol layer

Reply:
Thanks for guidance.
I noticed through testing that in Apache HttpComponents, responseTimeout and socketTimeout do not operate completely independently at execution time, responseTimeout effectively overrides the socket’s read timeout. My intention with this PR was to make that behavior clearer in the documentation.

Would the following revised wording be acceptable?